stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -148,11 +205,7 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
      */
     inline void join(const recursive_reducer& rhs) {
       sum_ += rhs.sum_;
-      if (args_adjoints_.size() != 0 && rhs.args_adjoints_.size() != 0) {
-        args_adjoints_ += rhs.args_adjoints_;
-      } else if (args_adjoints_.size() == 0 && rhs.args_adjoints_.size() != 0) {
-        args_adjoints_ = rhs.args_adjoints_;
-      }
+      args_adjoints_.noalias() += rhs.args_adjoints_;
```

Comment:
Oh were the ifs here earlier just not used at all? I don't think this needs an `.noalias`.

From: https://eigen.tuxfamily.org/dox/group__TopicAliasing.html

> Matrix multiplication is the only operation in Eigen that assumes aliasing by default, under the condition that the destination matrix is not resized. Thus, if matA is a squared matrix, then the statement matA = matA * matA; is safe. All other operations in Eigen assume that there are no aliasing problems, either because the result is assigned to a different matrix or because it is a component-wise operation.

Reply:
Could ask Steve or Tadej to clarify they know this stuff better

Reply:
But putting noalias there does not hurt, right?

I don't think the if stuff was used... when I read the TBB doc I figured that whenever a reduce worker is split off, then it will do some stuff and hence this delay init stuff should not be needed.

Reply:
Yeah doesn't hurt, but might as well clean up while we're here. I asked Steve in slack.